wsd/wopi/WopiStorage.cpp

```diff
@@ -784,12 +779,14 @@ void WopiStorage::uploadLocalFileToStorageAsync(const Authorization& auth, LockC
 
         LOG_DBG(wopiLog << " async upload request: " << httpRequest.header().toString());
 
+        _uploadHttpSession->setConnectFailHandler([asyncUploadCallback]() {
+            LOG_ERR("Cannot connect for uploading to wopi storage.");
+            asyncUploadCallback(AsyncUpload(AsyncUpload::State::Error,
+                            UploadResult(UploadResult::Result::FAILED, "Connection failed.")));
+        });
+
         // Make the request.
-        if (_uploadHttpSession->asyncRequest(httpRequest, socketPoll))
-        {
-            scopedInvokeCallback.setArg(
-                AsyncUpload(AsyncUpload::State::Running, UploadResult(UploadResult::Result::OK)));
-        }
+        _uploadHttpSession->asyncRequest(httpRequest, socketPoll);
```

Comment:
Should happen before line 789 .. (who?)

// Notify client via callback that the request is in progress...
asyncUploadCallback(
            AsyncUpload(AsyncUpload::State::Running, UploadResult(UploadResult::Result::OK)));

Reply:
Yes, seeing as there is the near duplicate alt pr of https://github.com/CollaboraOnline/online/pull/9839 but that retains the "State:Running" case, let me rebase and reuse that to reintroduce this